File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/payment/BuildPaymentRequestTest.java
package org.hyperledger.indy.sdk.payment;

import org.hyperledger.indy.sdk.InvalidStructureException;
import org.hyperledger.indy.sdk.payments.IncompatiblePaymentException;
import org.hyperledger.indy.sdk.payments.Payments;
import org.hyperledger.indy.sdk.payments.UnknownPaymentMethodException;
import org.junit.Test;

import java.util.concurrent.ExecutionException;

import static org.hamcrest.CoreMatchers.isA;

public class BuildPaymentRequestTest extends PaymentIntegrationTest {

	@Test
	public void testBuildPaymentRequestWorksForUnknownPaymentMethod() throws Exception {
		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(UnknownPaymentMethodException.class));

		Payments.buildPaymentRequest(wallet, DID_TRUSTEE, inputs, outputs, null).get();
	}

	@Test
	public void testBuildPaymentRequestWorksForEmptyInputs() throws Exception {
		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(InvalidStructureException.class));

		Payments.buildPaymentRequest(wallet, DID_TRUSTEE, emptyArray, outputs, null).get();
	}

	@Test
	public void testBuildPaymentRequestWorksForIncompatiblePaymentMethods() throws Exception {
		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(IncompatiblePaymentException.class));

		Payments.buildPaymentRequest(wallet, DID_TRUSTEE, incompatibleInputs, outputs, null).get();
	}
}

File: wrappers/dotnet/indy-sdk-dotnet-test/PaymentTests/BuildPaymentRequestTest.cs
using Hyperledger.Indy.PaymentsApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.PairwiseTests
{
    [TestClass]
    public class BuildPaymentRequestTest : PaymentIntegrationTestBase
    {
        [TestMethod]
        public async Task TestBuildPaymentRequestWorksForUnknownPaymentMethod()
        {
            var ex = await Assert.ThrowsExceptionAsync<UnknownPaymentMethodException>(() =>
                 Payments.BuildPaymentRequestAsync(wallet, DID_TRUSTEE, inputs, outputs, null)
             );
        }

        [TestMethod]
        public async Task TestBuildPaymentRequestWorksForEmptyInputs()
        {
            var ex = await Assert.ThrowsExceptionAsync<InvalidStructureException>(() =>
                 Payments.BuildPaymentRequestAsync(wallet, DID_TRUSTEE, emptyArray, outputs, null)
             );
        }

        [TestMethod]
        public async Task TestBuildPaymentRequestWorksForEmptyOutputs()
        {
            var ex = await Assert.ThrowsExceptionAsync<InvalidStructureException>(() =>
                 Payments.BuildPaymentRequestAsync(wallet, DID_TRUSTEE, inputs, emptyObject, null)
             );
        }

        [TestMethod]
        public async Task TestBuildPaymentRequestWorksForIncompatiblePaymentMethods()
        {
            var ex = await Assert.ThrowsExceptionAsync<IncompatiblePaymentMethodsException>(() =>
                 Payments.BuildPaymentRequestAsync(wallet, DID_TRUSTEE, incompatibleInputs, emptyObject, null)
             );
        }

        [TestMethod]
        public async Task TestBuildPaymentRequestWorksForInvalidInputs()
        {
            var ex = await Assert.ThrowsExceptionAsync<InvalidStructureException>(() =>
                 Payments.BuildPaymentRequestAsync(wallet, DID_TRUSTEE, invalidInputs, outputs, null)
             );
        }
    }
}
